soh/src/overlays/actors/ovl_Item_Etcetera/z_item_etcetera.c

```diff
@@ -249,20 +247,17 @@ void ItemEtcetera_Draw(Actor* thisx, GlobalContext* globalCtx) {
     s32 type = this->actor.params & 0xFF;
 
     if (gSaveContext.n64ddFlag) {
-        s32 randoGetItemId = GI_NONE;
+        GetItemEntry randoGetItem = (GetItemEntry)GET_ITEM_NONE;
         if (type == ITEM_ETC_ARROW_FIRE) {
-            randoGetItemId = Randomizer_GetItemIdFromKnownCheck(RC_LH_SUN, GI_ARROW_FIRE);
+            randoGetItem = Randomizer_GetItemFromKnownCheck(RC_LH_SUN, GI_ARROW_FIRE);
         } else if (type == ITEM_ETC_LETTER) {
-            randoGetItemId = Randomizer_GetItemIdFromKnownCheck(RC_LH_UNDERWATER_ITEM, GI_LETTER_RUTO);
+            randoGetItem = Randomizer_GetItemFromKnownCheck(RC_LH_UNDERWATER_ITEM, GI_LETTER_RUTO);
         }
 
-        if ((randoGetItemId >= GI_MINUET_OF_FOREST && randoGetItemId <= GI_DOUBLE_DEFENSE) ||
-            (randoGetItemId >= GI_STICK_UPGRADE_20 && randoGetItemId <= GI_NUT_UPGRADE_40)) {
-            EnItem00_CustomItemsParticles(&this->actor, globalCtx, randoGetItemId);
-        }
+        EnItem00_CustomItemsParticles(&this->actor, globalCtx, randoGetItem);
 
-        if (randoGetItemId != GI_NONE) {
-            this->giDrawId = Randomizer_GetItemModelFromId(randoGetItemId);
+        if (randoGetItem.itemId != RG_NONE) {
+            this->giDrawId = ABS(randoGetItem.gi) - 1;
```

Comment:
do we do this in other places? if so, could we macro it and add some comments next to the macro explaining why/how it works?

Reply:
Yes it's to counteract how the `GID` gets turned into the `gi` value by the `GET_ITEM` macro. It multiplies it by either 1 or -1 according to the chest animation value and then adds 1. So this takes the absolute value and then subtracts 1. I can macro/function it or store the GID separately on the GetItemEntry, think I'd prefer the macro/function approach.

Reply:
Addressed by 58c9f64a1689fbf3364beec0f362103b02c856a0